code/modules/hydroponics/soil.dm

```diff
@@ -0,0 +1,33 @@
+/obj/machinery/portable_atmospherics/hydroponics/soil
+	name = "soil"
+	icon = 'icons/obj/hydroponics.dmi'
+	icon_state = "soil"
+	density = 0
+	use_power = 0
+	draw_warnings = 0
+
+/obj/machinery/portable_atmospherics/hydroponics/soil/attackby(var/obj/item/O as obj, var/mob/user as mob)
```

Comment:
You should always inherit the arguments from the base proc definition:
`attackby(var/obj/item/W, var/mob/user)`

Also: `as X` does not do anything in arguments, except if it's to be used as a verb.